MapboxNavigation/NavigationMapView.swift

```diff
@@ -160,12 +160,23 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
                               contentFrame.minY + contentFrame.height * 0.5))
     }
     
+    open var routeController: RouteController? {
+        didSet {
+            resumeNotifications()
+        }
+    }
+    
     /**
      Determines whether the map should follow the user location and rotate when the course changes.
      - seealso: NavigationMapViewCourseTrackingDelegate
      */
     open var tracksUserCourse: Bool = false {
         didSet {
+            guard let _ = routeController, tracksUserCourse == true else {
+                print("NavigationMapView.routeController must non-nil to set tracksUserCourse to true.")
```

Comment:
`tracksUserCourse` is basically redundant to `routeController` now. We can move all the logic below into `routeController`’s setter, conditioned on whether `routeController` is non-nil.

Reply:
And we can rename `routeController` to something that makes its purpose clear, like `userCourseController` (which would suggest a protocol by that name based on RouteController’s interface).